crates/fuel-core/src/query/subscriptions/test.rs

```diff
@@ -85,6 +85,8 @@ fn squeezed_during_block_production() -> TransactionStatus {
 enum TxStatus {
     /// The transaction has been submitted
     Submitted,
+    /// The transaction has been preconfirmed, but the final status could still be different
+    Preconfirmed,
```

Comment:
Also, why doesn't the `Preconfirmed` have sub-data. 

It looks like there are places where the conversions don't allow it. I can't tell if that's a smell or if it's fine:
```rs
                Flow::Continue(not_final_status) => {
                    match not_final_status {
                        NotFinal::Submitted => out.push(Ok(TxStatus::Submitted)),
                        NotFinal::Preconfirmation => out.push(Ok(TxStatus::Preconfirmed)),
                    }
```

Reply:
I made the statuses more explicit in this commit: 2ad08a64a43d855329e37cefe1492bd07c8e20d3